src/OpenSpaceToolkit/Astrodynamics/Trajectory/Sequence.cpp

```diff
@@ -244,18 +257,33 @@ Sequence::Solution Sequence::solve(const State& aState) const
 
             segmentSolutions.add(segmentSolution);
 
+            // Segment condition check
             if (!segmentSolution.conditionIsSatisfied)
             {
                 BOOST_LOG_TRIVIAL(warning) << "Segment condition is not satisfied." << std::endl;
 
                 return {segmentSolutions, false};
             }
 
+            // Event condition check
+            if (anEventConditionSPtr != nullptr &&
+                anEventConditionSPtr->isSatisfied(segmentSolution.states.accessLast(), initialState))
+            {
+                BOOST_LOG_TRIVIAL(debug) << "Sequence Event condition is satisfied." << std::endl;
+
+                return {segmentSolutions, true};
+            }
+
             initialState = segmentSolution.states.accessLast();
         }
     }
 
-    return {segmentSolutions, true};
+    if (eventConditionIsSatisfied)
```

Comment:
Are both this and line 270 required?

I might be reading the flow wrong, but this looks like:
- if you get to the end of the sequence and a sequence Event Condition _existed_ but was **not** satisfied, return true.

It looks like nothing ever sets a value to `eventConditionIsSatisfied`. I.e. maybe a name like `sequenceEventConditionExists` is more appropriate. (maybe I'm blind and missing it somewhere)

Reply:
So in line 234 we set
`    bool eventConditionIsSatisfied = anEventConditionSPtr == nullptr;`
So basically, if the eventCondition is a nullptr, then automatically it's "satisfied" as there is no condition.
If there isn't one, then it's set as false, and appropriately updated.
I thought this was clever but it might make the code less readable, what do you think?

Reply:
I don't think I follow... I believe this bool is never updated based on whether the event condition is actually satisfied. It's only set based on whether the event condition exists. So like 281 is effectively:
`if (sequenceEventConditionExists){return {segmentSolutions, true};}`

_But_ if we've gotten to this point in the code, that means we're at the end of all of the segments. I.e. the sequence event condition was _not_ hit. So this condition is effectively "if there was a sequence event condition but we didn't hit it, return success".

I might totally be missing something...